serverless.yml

```diff
@@ -31,6 +31,7 @@ custom:
     api_method: POST
     api_path: ${env:API_PATH}
     authToken: ${self:service}/${env:STAGE}/auth_token
+    tokenArn: ${env:TOKEN_ARN}
```

Comment:
I don't believe we need this, but I could be wrong.

If we need the ARN value to the auth_token stored in SSM, we are already specifying it on line 79: `"arn:aws:ssm:#{AWS::Region}:#{AWS::AccountId}:parameter/${self:custom.cfg.project}/${self:custom.cfg.authToken}"` as a resource in another permission.

So in line 88 below, instead of saying `${self:custom.cfg.tokenArn}`, we can say `"arn:aws:ssm:#{AWS::Region}:#{AWS::AccountId}:parameter/${self:custom.cfg.project}/${self:custom.cfg.authToken}"`. I know it's longer and less readable, but ultimately I think it's less complex since we will not need to specify the ARN value in buildspec, and it is still based on existing variables and not hard-coded.

What do you think?

Reply:
less readable and more complex is not ideal, I think having a var in the buildspec is fine, one place to manage vars is attractive.
We should just replace that line you mentioned with the var instead.

Reply:
actually, what you propose isn't that difficult, let's discuss on Monday, as long as we maintain the restriction I think that's what matters most.

Reply:
As we talked on Slack, I don't really have a strong opinion on this. I just don't think it's necessary to access that as an env variable while it's readily available.

However, I am OK with leaving it in as an env variable in buildspec as an enforcement for the users to create the SSM parameter before deployment.

Reply:
Been thinking about this and decided I am OK with this change :)